pkg/reconciler/route/resources/cluster_ingress.go

```diff
@@ -86,19 +88,27 @@ func makeClusterIngressSpec(r *servingv1alpha1.Route, targets map[string]traffic
 	}
 }
 
-func routeDomains(targetName string, r *servingv1alpha1.Route) []string {
-	domains := []string{traffic.TagDomain(targetName, r.Status.Domain)}
+func routeDomains(ctx context.Context, targetName string, r *servingv1alpha1.Route) []string {
+	fullName, _ := domains.DomainNameFromTemplate(ctx, r, domains.SubdomainName(r, targetName)) // TODO(andrew-su): Handle the error here.
+
+	ruleDomains := []string{fullName}
+
+	// TODO(andrew-su): We are adding this for backwards compatibility.
+	deprecatedFullName := traffic.TagDomain(targetName, r.Status.Domain)
+	if fullName != deprecatedFullName {
+		ruleDomains = append(ruleDomains, deprecatedFullName)
+	}
```

Comment:
So we just delete this when we're ready?

Reply:
The plan for this was to remove it in 0.7 giving people time to migrate, as there may be users still relying on the old format

Reply:
SGTM, just confirming :)